test/Sentry.Tests/SentryClientTests.cs

```diff
@@ -322,106 +332,48 @@ public void CaptureEvent_SamplingNull_DropsEvent()
         Assert.Same(@event, received);
     }
 
-    [Fact]
-    public void CaptureEvent_Sampling50Percent_EqualDistribution()
+    [Theory]
+    [InlineData(0.25f)]
+    [InlineData(0.50f)]
+    [InlineData(0.75f)]
+    public void CaptureEvent_WithSampleRate_AppropriateDistribution(float sampleRate)
```

Comment:
I consolidated these three tests into one, and then simplified it.  I also had to add some retry due to randomness of results.